BeagleBone_black/i2c/i2c.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * i2c.hpp
+ *
+ * Created: 10/02/2018 10:18:58
+ * Authors: M. Kristien, E. van Woerkom
+ */ 
+
+
+#ifndef I2C_HPP_
+#define I2C_HPP_
+
+namespace i2c {
+
+struct Channel;
+struct Context;
+class Driver;
+
+// Single instance of R/W periodic communications channel with a slave i2c device
+struct Channel {
+  uint8_t address;
+  char direction;
+  // 3 direction possibilities: W(rite), R(ead), G(et),
+  // with get being a write-read operation to retrieve register data,
+  // read is most likely never used, get often.
+  char* readbuffer, writebuffer; // Pointers to be left null if no read or write takes place
+  int readbuflength, writebuflength;
+  int period; // Number of microseconds to wait to start a new reading
+  int* status; // If something goes wrong, this must be set using || (or) so as not to remove unreseted errors
+  int* error_counter, success_counter; // To be incremented after every failure and success
+  void *callbackfunction(Channel* current_channel, Driver current_device);
```

Comment:
A member function in a struct? Maybe this is a sensible design but please give it a second thought and see if it really is more sensible than having a class instead of struct or not having the function in here.